### ISSUE TITLE
Allow random orderby in post queries

### ISSUE THREAD

**Issue by jeherve** [WP-API/WP-API#2926]
I'm having some trouble querying a list of random posts with the most recent version of the WP REST API.

Up until last month, one could use `rand` with the `orderby` [WP Query parameter][1] to retrieve posts in a random order, like so:

`/wp-json/wp/v2/posts?filter[orderby]=rand`

However, we recently got rid of that `filter` key structure in #2799. The request above would now look like this:

`wp-json/wp/v2/posts?orderby=rand`

That request won't return random posts, though:

```json
{
    "code":"rest_invalid_param",
    "message":"Invalid parameter(s): orderby",
    "data":{
        "status":400,
        "params":{
            "orderby":"orderby is not one of date, relevance, id, include, title, slug."
        }
    }
}
```

That's because the list of supported query options in the WP REST API isn't the same as for regular WP queries. Only 6 query args are allowed [here][2].

I looked for a filter allowing me to alter the list of allowed query arguments, but I couldn't find anything. I know we have `rest_query_vars` to edit the list of query vars, but I didn't find anything equivalent for query args.

I know I could alter the whole query and force `rand` with the `rest_post_query` filter, but that's not what I'm looking for here.

Am I missing something obvious here?

Thank you!

Related: #2906

  [1]: https://codex.wordpress.org/Class_Reference/WP_Query#Order_.26_Orderby_Parameters
  [2]: https://github.com/WP-API/WP-API/blob/develop/lib/endpoints/class-wp-rest-posts-controller.php#L1866-L1882

**Comment by adamsilverstein** [comment]
@jeherve You are right, it looks like the current accepted arguments here don't include several from WP_Query that might be useful for posts: 'none', 'modified', 'parent', 'rand' and 'meta_value_num'. Also, I don't see a filter to easily alter these.

Not sure about the history behind why these options are missing - it would be nice to support them all by default and provide a way for developers to alter the accepted args via a filter.

**Comment by jeherve** [comment]
> Not sure about the history behind why these options are missing - it would be nice to support them all by default

I assume some were removed for security reasons, others for performance reasons, but I couldn't find any discussion about it here. I'm watching [this Core ticket](https://core.trac.wordpress.org/ticket/38693), where I guess we can discuss adding more accepted arguments, but a filter would work for me too, as you mentioned.

**Comment by nylen** [comment]
These extra `orderby` options were unintentionally removed as a side effect of enabling strict validation for `enum` arguments.

@jeherve can you close this issue and move discussion (and the list of proposed additions) to that Trac ticket? I don't see why we couldn't get this regression fixed in 4.7.

**Comment by jeherve** [comment]
👍 Closing in favor of https://core.trac.wordpress.org/ticket/38693
